pylabrobot/heating_shaking/hamilton.py

```diff
@@ -61,20 +61,25 @@ async def shake(
     assert direction in [0, 1], "Direction must be 0 or 1"
     assert 500 <= acceleration <= 10_000, "Acceleration must be between 500 and 10_000"
 
-    if not self.lock_initialized:
-      await self._initialize_lock()
-    await self._move_plate_lock(PlateLockPosition.UNLOCKED)
     await self._start_shaking(direction=direction, speed=int_speed, acceleration=acceleration)
 
   async def stop_shaking(self):
     """ Shaker `stop_shaking` implementation. """
     await self._stop_shaking()
     await self._wait_for_stop()
-    await self._move_plate_lock(PlateLockPosition.LOCKED)
 
   async def _move_plate_lock(self, position: PlateLockPosition):
+    if not self.lock_initialized:
+      await self._initialize_lock()
```

Comment:
i thought this was only required on `setup`?

Reply:
We may not want to run it on setup, so this is more flexible, but either way.

Reply:
I generally try to initialize everything on setup to get super predictable protocols